### PR TITLE
Add initial bridge implementation

### PR DESCRIPTION
**Pull request by dguenther**
## Summary

Adds a `/bridges` route with APIs for fetching bridgeable assets, fetching destination networks for those assets, creating transaction outputs, and checking transaction status.

Fixes IFL-2984
Fixes IFL-2986
Fixes IFL-2987
Fixes IFL-2988
Fixes IFL-2994

## Testing Plan

I wrote tests for the ChainportService, but since it hits the live API I've skipped them. I think they're still useful to have committed in case we need to update the implementation. 

## Breaking Change

Is this a breaking change? If yes, add notes below on why this is breaking and
what additional work is required, if any.

```
[ ] Yes
```

### CODE REVIEW

File: src/bridges/chainport.service.ts [hunk 1]
```diff
@@ -0,0 +1,345 @@
...
+    return firstValueFrom(
+      this.httpService.get<T>(url).pipe(
+        catchError((e: AxiosError) => {
+          this.logger.error(
+            `Chainport Error: ${url} - Status Code ${
+              e.status ?? 'Unknown'
+            } - ${JSON.stringify(e.response?.data)}`,
+            new Error().stack ?? '',
+          );
+          throw e;
```
Comment (rohanjadvani) [inline src/bridges/chainport.service.ts:146]: NestJS has `BadGatewayException` if we want to throw a 502 here. I think if we see that in the responses then we can check the logs accordingly.

File: src/bridges/chainport.service.ts [hunk 2]
```diff
@@ -0,0 +1,345 @@
...
+
+    const validateResult = chainportNetworkArraySchema.validate(
+      Object.values(meta.cp_network_ids),
+      {
+        stripUnknown: true,
+      },
+    );
+
+    if (validateResult.error) {
+      throw new Error(
```
Comment (rohanjadvani) [inline src/bridges/chainport.service.ts:183]: Same comment here for `BadGatewayException`.

File: src/bridges/chainport.service.ts [hunk 3]
```diff
@@ -0,0 +1,345 @@
...
+
+    const validateResult = chainportTokenArraySchema.validate(
+      result.data.verified_tokens,
+      {
+        stripUnknown: true,
+      },
+    );
+
+    if (validateResult.error) {
+      throw new Error(
```
Comment (rohanjadvani) [inline src/bridges/chainport.service.ts:215]: Same comment re `BadGatewayException`

File: src/bridges/chainport.service.ts [hunk 4]
```diff
@@ -0,0 +1,345 @@
...
+      if (!sourceToken) {
+        throw new NotFoundException();
+      }
+
+      networkList = sourceToken.target_networks.flatMap((n) => {
+        const network = metaResult.cp_network_ids[n.toString()];
+        if (!network) {
+          this.logger.error(
+            `Network ${n} for token ${tokenId} not found in meta`,
+            new Error().stack ?? '',
```
Comment (rohanjadvani) [inline src/bridges/chainport.service.ts:249]: Does `new Error().stack` always return some stack we can use? Or should we just use ''`?

File: src/bridges/chainport.service.ts [hunk 5]
```diff
@@ -0,0 +1,345 @@
...
+    url: string,
+  ): Promise<AxiosResponse<T, unknown>> {
+    return firstValueFrom(
+      this.httpService.get<T>(url).pipe(
+        catchError((e: AxiosError) => {
+          this.logger.error(
+            `Chainport Error: ${url} - Status Code ${
+              e.status ?? 'Unknown'
+            } - ${JSON.stringify(e.response?.data)}`,
+            new Error().stack ?? '',
```
[the remark this answers is not in the document]
Comment (dguenther) [reply src/bridges/chainport.service.ts:144]: Good idea 👍

File: src/bridges/chainport.service.ts (hunk 4 above)
Comment (dguenther) [reply src/bridges/chainport.service.ts:249]: Yep! It returns the stack at the current location

**Review by rohanjadvani — APPROVED** [review]

**Pull request merged**